### PR TITLE
Fix some issues

### PR DESCRIPTION
**Pull request by asmeurer**
Some issues that came up with testing pytorch against the 2022 spec (this doesn't address any of the issues I opened).

### CODE REVIEW

File: array_api_tests/test_array_object.py [hunk 1]
```diff
@@ -250,8 +250,8 @@ def make_param(method_name: str, dtype: DataType, stype: ScalarType) -> Param:
 @pytest.mark.parametrize(
     "method_name, dtype, stype",
     [make_param("__bool__", xp.bool, bool)]
-    + [make_param("__int__", d, int) for d in dh.all_int_dtypes]
-    + [make_param("__index__", d, int) for d in dh.all_int_dtypes]
+    + [make_param("__int__", d, int) for d in dh._filter_stubs(*dh.all_int_dtypes)]
+    + [make_param("__index__", d, int) for d in dh._filter_stubs(*dh.all_int_dtypes)]
```
Comment (asmeurer) [inline array_api_tests/test_array_object.py:254]: Let me know if there's a better way to do this. I'm not sure if this internal function is supposed to be used outside of the dtype_helpers module. I'm a little unclear why `all_int_dtype` isn't just filtered to begin with?

Comment (honno) [reply array_api_tests/test_array_object.py:254]: Alternative approach which skips undefined stubs

```diff
def test_scalar_casting(method_name, dtype, stype, data):
+    if isinstance(dtype, xp._UndefinedStub):
+        pytest.skip(f"{dtype} not in xp namespace")
```

I could go either way with this. I like the idea of having the test case for `uint{16,32,64}` just exist as it is expected behaviour in the spec afterall, but I also get its noisy.

> I'm a little unclear why `all_int_dtype` isn't just filtered to begin with?

IIRC it's for the above reason, so it exists for such test cases and we can subsequently skip. Again, could go either way on that now.

**Review by honno — APPROVED** [review]
LGTM!

File: array_api_tests/test_array_object.py (hunk 1 above)
Comment (asmeurer) [reply array_api_tests/test_array_object.py:254]: Oh yeah skipping is probably better.

**Pull request merged**